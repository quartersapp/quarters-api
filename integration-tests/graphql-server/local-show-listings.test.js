/* eslint-env jest */

const request = require('supertest')
const app = require('lib/app').listen()
const {
  Artist,
  City,
  Show,
  Venue,
  ShowArtist
} = require('lib/db/models')
const { truncateModel } = require('test-helpers')

let city

beforeEach(async () => {
  await Promise.all([
    ShowArtist,
    Artist,
    City,
    Show,
    Venue
  ].map(truncateModel))

  city = await City.query()
    .insert({
      name: 'Toronto',
      googlePlaceId: 'place-id'
    })
    .returning('*')

  const venue = await Venue.query()
    .insert({
      name: 'Horseshoe Tavern',
      cityId: city.id,
      googlePlaceId: 'place-id'
    })
    .returning('*')

  const show1 = await Show.query()
    .insert({
      venueId: venue.id,
      day: '2018-05-01'
    })
    .returning('*')

  const show2 = await Show.query()
    .insert({
      venueId: venue.id,
      day: '2018-05-05'
    })
    .returning('*')

  const andyShauf = await Artist.query()
    .insert({
      name: 'Andy Shauf',
      cityId: city.id
    })
    .returning('*')

  const bornRuffians = await Artist.query()
    .insert({
      name: 'Born Ruffians',
      cityId: city.id
    })
    .returning('*')

  const localNatives = await Artist.query()
    .insert({
      name: 'Local Natives',
      cityId: city.id
    })
    .returning('*')

  await ShowArtist.query().insert({
    showId: show1.id,
    artistId: andyShauf.id,
    position: 0
  })

  await ShowArtist.query().insert({
    showId: show1.id,
    artistId: bornRuffians.id,
    position: 1
  })

  await ShowArtist.query().insert({
    showId: show2.id,
    artistId: localNatives.id,
    position: 1
  })
})

it("can be queried for a city's show listings", async () => {
  const query = `
    {
      showsForCity (cityId: ${city.id}) {
        day
        venue { name }
        artists { name }
      }
    }
  `

  const { body } = await request(app)
    .post('/graphql')
    .send({ query })

  expect(body).toMatchSnapshot()
})
