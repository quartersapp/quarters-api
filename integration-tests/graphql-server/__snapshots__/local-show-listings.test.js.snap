// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can be queried for a city's show listings 1`] = `
Object {
  "data": Object {
    "showsForCity": Array [
      Object {
        "artists": Array [
          Object {
            "name": "Andy Shauf",
          },
          Object {
            "name": "Born Ruffians",
          },
        ],
        "day": "2018-05-01",
        "venue": Object {
          "name": "Horseshoe Tavern",
        },
      },
      Object {
        "artists": Array [
          Object {
            "name": "Local Natives",
          },
        ],
        "day": "2018-05-05",
        "venue": Object {
          "name": "Horseshoe Tavern",
        },
      },
    ],
  },
}
`;
